aiida_common_workflows/common/visualization/dissociation.py

```diff
@@ -18,6 +18,13 @@ def get_dissociation_plot(
     :param unit_distance: unit of volume, default is [Å^3].
     :param unit_energy: unit of energy, default is [eV].
     """
+    if len(distances) != len(energies):
+        raise ValueError('`distances` and `energies` are not of the same length.')
+    if not all([isinstance(d, float) for d in distances]):
```

Comment:
```suggestion
    if any(not isinstance(d, float) for d in distances):
```
apologies for the nitpick, but this way you don't have an intermediate list-comprehension and so is more efficient. Shouldn't matter much in practice, but maybe good idea to adopt everywhere consistently.
The paradigm:
```
all(e for e in x)
```
is identical in function to
```
all([e for e in x])
```
except in the latter, the full list comprehension is built first and then the resulting list is applied to `all`. Whereas in the former a generator is used directly for `all` and will stop as soon as it is violated. The same goes for `any`.

Please also adjust the `get_eos_plot` accordingly

Reply:
> apologies for the nitpick

Not at all! This is solid info. Thanks for taking the time to explain the difference.